openedx/core/djangoapps/enterprise/models.py

```diff
@@ -0,0 +1,163 @@
+Models for Enterprise features
+"""
+from django.contrib.auth.models import User
+from django.db import models
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+from django.utils.translation import ugettext_lazy as _
+from django_extensions.db.models import TimeStampedModel
+
+
+class EnterpriseCustomerManager(models.Manager):
+    """
+    Model manager for Enterprise Customer model.
+
+    Filters out inactive Enterprise Customers, otherwise works the same as default model manager.
+    """
+    # This manager filters out some records, hence according to the Django docs it must not be used
+    # for related field access. Although False is default value, it still makes sense to set it explicitly
+    # https://docs.djangoproject.com/en/1.10/topics/db/managers/#base-managers
+    use_for_related_fields = False
+
+    def get_queryset(self):
+        """
+
+        :return:
+        """
+        return super(EnterpriseCustomerManager, self).get_queryset().filter(active=True)
+
+
+class EnterpriseCustomer(models.Model):
+    """
+    Enterprise Customer is an organization or a group of people that "consumes" courses: members of Enterprise
+    Customer take courses on the edX platform.
+
+    Enterprise Customer might be providing certain benefits to their members, like discounts to paid course
+    enrollments, and also might request (or require) sharing learner results with them.
+
+    Fields:
+        [implicit] id (IntegerField): EnterpriseCustomer identifier, PRIMARY KEY
+        name (CharField): Enterprise Customer name
+        logo (ImageField):  Enterprise Customer logo
+        data_sharing_policy (CharField): stores data sharing policy: either no data sharing,
+                                         requested data sharing (learners are allowed to opt-in) or required data
+                                         sharing (learners must share their data)
+        sso_providers (CharField): list of comma-separated SSO provider slugs. Providers in this list should link
+                                    users registered with that provider with Enterprise Customer
+        active (BooleanField): used to mark inactive Enterprise Customers - implements "soft delete" pattern
+        members (ManyToMany[User]): list of Enterprise Customer members.
+    """
+    NO_DATA_SHARING = "no_sharing"
+    REQUEST_DATA_SHARING = "request"
+    REQUIRE_DATA_SHARING = "require"
+
+    class Meta(object):
+        app_label = "enterprise"
+        verbose_name = "Enterprise Customer"
+
+    objects = EnterpriseCustomerManager()
+    all_objects = models.Manager()
+
+    name = models.CharField(max_length=500, blank=False, null=False, help_text=_(u"Enterprise Customer name."))
+    logo = models.ImageField(
+        upload_to="enterprise_logos",
+        help_text=_(u"Please add only .PNG files for logo images."),
+        null=True, blank=True, max_length=255
+    )
+    data_sharing_policy = models.CharField(
```

Comment:
@mattdrayer This field overlaps with Data Sharing Consent models, but it might be a better approach to store the data on enterprise customer, rather than on the SSO provider.